cli/command/image/trust.go

```diff
@@ -301,8 +299,11 @@ func trustedPull(ctx context.Context, cli *command.DockerCli, repoInfo *registry
 		}
 
 		// If reference is not trusted, tag by trusted reference
-		if !r.reference.HasDigest() {
-			tagged, err := reference.WithTag(repoInfo, r.reference.String())
+		// TODO(aaronl): Is this code really necessary? Will notary ever
```

Comment:
👍  Notary should not give you a digest as the name, unless someone specifically added a digest as the name using the notary CLI (e.g. this is not something that would have happened via docker push), and it seems fair to not support that use case (especially if as @dmcgowan says if it's a digest it gets shunted down a different path anyway)